File: debuglogger/src/main/java/io/github/japskiddin/debuglogger/LogEvent.java
package io.github.japskiddin.debuglogger;

import androidx.annotation.NonNull;
import java.text.SimpleDateFormat;
import java.util.Locale;

public class LogEvent implements Comparable<LogEvent> {
  private final long time;
  private final String tag;
  private final String text;

  public LogEvent(@NonNull String tag, String text) {
    this.time = System.currentTimeMillis();
    this.tag = tag;
    this.text = text;
  }

  public String getTag() {
    return tag;
  }

  public long getTime() {
    return time;
  }

  public String getText() {
    return text;
  }

  @Override
  public int compareTo(LogEvent logEvent) {
    return this.time > logEvent.time ? 1 : -1;
  }

  @NonNull @Override public String toString() {
    SimpleDateFormat sdf = new SimpleDateFormat("HH:mm:ss.SSS", Locale.getDefault());
    return tag + " - " + sdf.format(time) + " - " + text;
  }
}

File: debuglogger/src/main/java/io/github/japskiddin/debuglogger/LogEvent.kt
package io.github.japskiddin.debuglogger

import java.text.SimpleDateFormat
import java.util.Locale

class LogEvent(private val type: Int, private val tag: String, private val text: String) :
  Comparable<LogEvent> {
  val time: Long = System.currentTimeMillis()

  private val logType: String
    get() = when (type) {
      LOG_INFO -> "INFO"
      LOG_WARN -> "WARN"
      LOG_ERROR -> "ERROR"
      LOG_DEBUG -> "DEBUG"
      else -> "INFO"
    }

  override fun compareTo(other: LogEvent): Int {
    return if (time > other.time) 1 else -1
  }

  override fun toString(): String {
    val sdf = SimpleDateFormat("HH:mm:ss.SSS", Locale.getDefault())
    return sdf.format(time) + " - " + logType + ": " + tag + " / " + text
  }

  companion object {
    const val LOG_INFO = 0
    const val LOG_WARN = 1
    const val LOG_ERROR = 2
    const val LOG_DEBUG = 3
  }
}